controllers/chat/sendSignedMessage.js

```diff
@@ -74,11 +84,26 @@ const sendSignedMesage = async (req, res) => {
     if (createdChannel?.channel?.is_channel_blocked)
       return res.status(403).json(responseError('Channel is blocked'));
 
-    const chat = await channel.sendMessage({
+    const currentMessageType = determineMessageType(messageType, attachments);
+    let baseMessage = {
       user_id: req.userId,
       attachments,
+      message_type: currentMessageType,
       text: message
-    });
+    };
+
+    if (messageType === MESSAGE_TYPE.REPLY) {
+      baseMessage = await processReplyMessage({
+        res,
+        baseMessage,
+        channelId,
+        client,
+        messageType,
+        replyMessageId
+      });
+    }
+
+    const chat = await channel.sendMessage(baseMessage);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This new block of code introduces handling for different message types, specifically reply messages. The `determineMessageType` function is used to determine the type of the message based on the `messageType` and `attachments` parameters. If the `messageType` is a reply, the `processReplyMessage` function is called to process the reply message. This adds more flexibility to the chat functionality by allowing different types of messages. However, ensure that the `processReplyMessage` function handles all edge cases and errors correctly to avoid any potential issues.


<!-- This is an auto-generated comment by OSS CodeRabbit -->